core/src/test/java/com/yahoo/tagchowder/ParserTest.java

```diff
@@ -51,6 +53,98 @@ public void testSampleHtml() throws IOException, SAXException {
         parser.parse(inSource);
     }
 
+    /**
+     * Parse an sample AMP html.
+     *
+     * @throws IOException IOException
+     * @throws SAXException SAXException
+     */
+    @Test
+    public void testSampleAMPHtml() throws IOException, SAXException {
+        final String html = getSampleHtml("amphtml.txt");
+        final Parser parser = new Parser();
+        final CustomHandler customHandler = new CustomHandler();
+        parser.setContentHandler(customHandler);
+        parser.setErrorHandler(customHandler);
+        parser.setFeature(Parser.AMP_VALIDATION_FEATURE, true);
+        parser.setDefaultBufferSize(html.length());
+        final InputSource inSource = new InputSource(new StringReader(html));
+        parser.parse(inSource);
+
+        List<ParsedHtmlTag> parsedHtmlTagList = customHandler.getParsedHtmlTags();
+        Assert.assertEquals(parsedHtmlTagList.size(), 21);
+
+        ParsedHtmlTag parsedHtmlTag = parsedHtmlTagList.get(0);
+        Assert.assertEquals(parsedHtmlTag.lowerName(), "html");
+        Assert.assertTrue(parsedHtmlTag.hasAttribute("⚡4email"));
+
+        parsedHtmlTag = parsedHtmlTagList.get(7);
+        Assert.assertEquals(parsedHtmlTag.lowerName(), "p");
+        Assert.assertTrue(parsedHtmlTag.hasAttribute("[[text]]"));
+
+        parsedHtmlTag = parsedHtmlTagList.get(12);
+        Assert.assertEquals(parsedHtmlTag.lowerName(), "p");
+        Assert.assertFalse(parsedHtmlTag.hasAttribute("{[text}"));
```

Comment:
Please add these check for all 9 santize attributes name in your amphtml.txt

Reply:
Done